core/src/main/java/jenkins/security/ResourceDomainConfiguration.java

```diff
@@ -112,9 +114,11 @@ private FormValidation checkUrl(String resourceRootUrlString, boolean allowOnlin
 
         URL resourceRootUrl;
         try {
-            resourceRootUrl = new URL(resourceRootUrlString);
+            resourceRootUrl = new URI(resourceRootUrlString).toURL();
         } catch (MalformedURLException ex) {
             return FormValidation.error(Messages.ResourceDomainConfiguration_Invalid());
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
```

Comment:
this is wrong and regresses functionality - the error needs to be user visible so they can fix it.

Reply:
@jtnord  I have replaced
```
        } catch (URISyntaxException e) {
            throw new RuntimeException(e);
```
with
```
        } catch (URISyntaxException e) {
            throw new URISyntaxException(e.getReason(), e.getInput());
```
Thus, used visibility is covered now for the specific exception i.e. `URISyntaxException`